src/battle/partner/lakilester.c

```diff
@@ -19,7 +44,244 @@ ApiStatus func_80238000_70BD10(Evt* script, s32 isInitialCall) {
+            target = get_actor(partner->targetActorID);
+            part = get_actor_part(target, partner->targetPartIndex);
+            partnerState->goalPos.x += part->unk_75;
+            partnerState->goalPos.y += part->unk_76;
+            partnerState->distance = dist2D(partnerState->currentPos.x,
+                                            partnerState->currentPos.y,
+                                            partnerState->goalPos.x,
+                                            partnerState->goalPos.y);
+            partnerState->speed = 0.0f;
+            partnerState->unk_30.x = 0.0f;
+            partnerState->unk_30.y = 0.0f;
+            D_8023D290 = 0;
+            D_8023D294 = 0;
+            D_8023D298 = 30;
+            D_8023D2A0 = 0.0f;
+            D_8023D2A4 = 0.0f;
+            D_8023D2A8 = 0.0f;
+            D_8023D2AC = 0.0f;
+            D_8023D29C = 0;
+            D_8023D2B0 = 3.0f;
+            D_8023D2B4 = -1.0f;
+            partnerState->unk_24 = ((part->size.y + part->size.x) / 2) / 24.0;
+            hud_element_set_scale(D_8023D28C, partnerState->unk_24 * target->scalingFactor);
+            D_8023D2C8 = 1;
+            script->functionTemp[0] = 1;
+            break;
+        case 1:
+            if (battleStatus->stickX != 0 || battleStatus->stickY != 0) {
+                D_8023D298 = 0;
+            }
+            if (D_8023D298 != 0) {
+                D_8023D298--;
+            } else {
+                D_8023D298 = 0x50;
+                hud_element_set_script(D_8023D2BC, HES_StickTapRight);
+                sfx_play_sound_at_position(0x312, 0, 0.0f, 0.0f, 0.0f);
+                script->functionTemp[0] = 2;
+            }
+            break;
+        case 2:
+            dpadY = battleStatus->dpadY;
+            if (dpadY > 0x50) {
+                dpadY = 0x50;
+            }
+            dpadX = battleStatus->dpadX;
+            if (dpadY < 35) {
+                dpadY *= sin_rad(DEG_TO_RAD(dpadY * 2));
+            }
+            temp_f8 = D_8023D2A0;
+            temp_f2 = temp_f8 + D_8023D2A4 + D_8023D2A8;
+            D_8023D2A0 = dpadY;
+            D_8023D2A8 = D_8023D2A4;
+            temp_f10 = D_8023D2A0 - (temp_f2 / 3.0f);
+            D_8023D2A4 = temp_f8;
+            if (temp_f10 > 0) {
+                D_8023D2AC = temp_f10;
+                if (partner->state.speed != 0.0f) {
+                    partner->state.speed += (D_8023D2AC / 8.5) / 3.0;
+                } else {
+                    partner->state.speed += (D_8023D2AC / 8.5) * 0.5;
+                }
+                D_8023D2B4 = dpadX;
+            } else {
+                D_8023D2AC = temp_f10;
+            }
+            if (!(dpadY < 31)) {
+                D_8023D2B4 = dpadX;
+            }
+            if (partnerState->speed > 6.7) {
+                partnerState->speed = 6.7f;
+            }
+            partnerState->speed = partnerState->speed - (f32) ((2.0 * (f64) (0x28 - dpadY)) / 40.0);
```

Comment:
casts needed?

Reply:
the first one is, but I'll remove the second